mantle/kola/tests/coretest/core.go

```diff
@@ -37,22 +43,58 @@ var offServices = []string{
 	"tcsd.service",
 }
 
+var nativeFuncs = map[string]register.NativeFuncWrap{
+	"PortSSH":        register.CreateNativeFuncWrap(TestPortSsh),
+	"DbusPerms":      register.CreateNativeFuncWrap(TestDbusPerms),
+	"ServicesActive": register.CreateNativeFuncWrap(TestServicesActive),
+	"ReadOnly":       register.CreateNativeFuncWrap(TestReadOnlyFs),
+	"Useradd":        register.CreateNativeFuncWrap(TestUseradd),
+	"MachineID":      register.CreateNativeFuncWrap(TestMachineID),
+	"RHCOSGrowpart":  register.CreateNativeFuncWrap(TestRHCOSGrowfs, []string{"fcos"}...),
+	"FCOSGrowpart":   register.CreateNativeFuncWrap(TestFCOSGrowfs, []string{"rhcos"}...),
+}
+
 func init() {
+
 	register.RegisterTest(&register.Test{
 		Name:        "basic",
 		Description: "Verify basic functionalities like SSH, systemd services, useradd, etc.",
 		Run:         LocalTests,
 		ClusterSize: 1,
-		NativeFuncs: map[string]register.NativeFuncWrap{
-			"PortSSH":        register.CreateNativeFuncWrap(TestPortSsh),
-			"DbusPerms":      register.CreateNativeFuncWrap(TestDbusPerms),
-			"ServicesActive": register.CreateNativeFuncWrap(TestServicesActive),
-			"ReadOnly":       register.CreateNativeFuncWrap(TestReadOnlyFs),
-			"Useradd":        register.CreateNativeFuncWrap(TestUseradd),
-			"MachineID":      register.CreateNativeFuncWrap(TestMachineID),
-			"RHCOSGrowpart":  register.CreateNativeFuncWrap(TestRHCOSGrowfs, []string{"fcos"}...),
-			"FCOSGrowpart":   register.CreateNativeFuncWrap(TestFCOSGrowfs, []string{"rhcos"}...),
-		},
+		NativeFuncs: nativeFuncs,
+	})
+
+	register.RegisterTest(&register.Test{
+		Name:          "basic.uefi",
+		Description:   "Verify basic functionalities like SSH, systemd services, useradd, etc, with uefi enabled",
+		Firmwares:     []string{uefi},
+		Platforms:     []string{"qemu"},
+		Architectures: []string{"x86_64"},
+		Run:           uefiWithBasicTests,
+		ClusterSize:   0,
+		NativeFuncs:   nativeFuncs,
+	})
+
+	register.RegisterTest(&register.Test{
+		Name:          "basic.uefi.secure",
+		Description:   "Verify basic functionalities like SSH, systemd services, useradd, etc, with uefi enabled",
+		Firmwares:     []string{uefiSecure},
+		Platforms:     []string{"qemu"},
+		Architectures: []string{"x86_64"},
+		Run:           uefiSecureWithBasicTests,
+		ClusterSize:   0,
+		NativeFuncs:   nativeFuncs,
+	})
+
+	register.RegisterTest(&register.Test{
+		Name:          "basic.nvme",
```

Comment:
OK right. I know I said in https://github.com/coreos/coreos-assembler/pull/3652#pullrequestreview-1723474562 to just match the coverage we currently have, and you've correctly done that here. But actually looking at what `cmd-kola` was doing, I'm surprised to learn that on x86_64, it's *not* testing the not-NVMe version, which is really surprising. So I think we want `basic` and `basic.nvme`. Both of those should be able to run on all arches. The former should additionally be runnable on all platforms.

Reply:
hmm, so would this not be covered by the already existing basic test? https://github.com/coreos/coreos-assembler/blob/a8998890db5a8a091dd13242aeb555ef53a35bbb/mantle/kola/tests/coretest/core.go#L59

Reply:
Yes, you're right. I got confused here and thought this whole hunk was replacing that bit too.

Should be able to drop the arch limitation on this one though.

Reply:
No worries, thank you!